cnc_ctrl_v1/Motor.cpp

```diff
@@ -83,42 +83,37 @@ void Motor::write(int speed){
     Sets motor speed from input. Speed = 0 is stopped, -255 is full reverse, 255 is full ahead.
     */
     if (_attachedState == 1){
-        
-        //linearize the motor
-        //speed = _convolve(speed);
-        
-        //set direction range is 0-180
-        if (speed > 0){
-            digitalWrite(_pin1 , HIGH);
-            digitalWrite(_pin2 , LOW );
-            speed = speed;
-        }
-        else if (speed == 0){
-            speed = speed;
-        }
-        else{
-            digitalWrite(_pin1 , LOW);
-            digitalWrite(_pin2 , HIGH );
-            speed = speed;
-        }
-        
-        //enforce range
-        speed = constrain(speed, -255, 255);
-        
-        _lastSpeed = speed; //saves speed for use in additive write
-        
-        speed = abs(speed); //remove sign from input because direction is set by control pins on H-bridge
-        
-        int pwmFrequency = round(speed);
-        
-        if(_pwmPin == 12){
-            pwmFrequency = map(pwmFrequency, 0, 255, 0, 1023);  //Scales 0-255 to 0-1023
-            Timer1.pwm(2, pwmFrequency);  //Special case for pin 12 due to timer blocking analogWrite()
-        }
-        else{
-            analogWrite(_pwmPin, pwmFrequency);
-        }
-        
+      speed = constrain(speed, -255, 255);
+      _lastSpeed = speed; //saves speed for use in additive write
+      bool forward = (speed > 0);
+      speed = abs(speed); //remove sign from input because direction is set by control pins on H-bridge   
+
+      if(_pwmPin == 12){
+          if (forward){
+              analogWrite(_pin1 , abs(speed));
```

Comment:
speed is already in + terms here, no need for the extra abs() call.  The same is true in the additional calls below.

Reply:
" add a note that it applies to specific board versions." Where to put this note? Dows it belong in  Motor.cpp, or in the description of the PR?

Reply:
You are right, I think your added comments are sufficient.